Nice inputs and buttons

akellbl4 opened the pull request:
Hi there,
Sorry, I know it might be better if I had solved some issues. (And I am ready for it)

This is PR with small UI changes connected with inputs and buttons.

1. I updated login forms with clean fields and restyled buttons.
<details>
  <summary>Screenshots</summary>

   ![Screenshot 2019-12-30 at 16 40 02](https://user-images.githubusercontent.com/2330682/71584480-c52afd00-2b23-11ea-9bc3-e7d071196971.png)
   ![Screenshot 2019-12-30 at 16 40 07](https://user-images.githubusercontent.com/2330682/71584483-c78d5700-2b23-11ea-9e6c-3e9c80c0c3ef.png)
</details>


2. I updated the style of a button in the comment
<details>
  <summary>Screenshots</summary>

  ![Screenshot 2019-12-30 at 16 40 34](https://user-images.githubusercontent.com/2330682/71584568-28b52a80-2b24-11ea-895b-78f57118bfe6.png)
</details>

3. And I used these UI elements for all different places (login buttons, dropdowns, buttons under a comment
<details>
  <summary>Screenshots</summary>

  ![Screenshot 2019-12-30 at 16 40 44](https://user-images.githubusercontent.com/2330682/71584604-60bc6d80-2b24-11ea-85f9-474bc115eb2e.png)
![Screenshot 2019-12-30 at 16 40 52](https://user-images.githubusercontent.com/2330682/71584605-60bc6d80-2b24-11ea-9958-b74371c800a5.png)
![Screenshot 2019-12-30 at 16 41 04](https://user-images.githubusercontent.com/2330682/71584606-60bc6d80-2b24-11ea-991d-0e1fbe49cfa2.png)
![Screenshot 2019-12-30 at 16 41 17](https://user-images.githubusercontent.com/2330682/71584607-61550400-2b24-11ea-8df1-de2194b731f1.png)
![Screenshot 2019-12-30 at 16 41 23](https://user-images.githubusercontent.com/2330682/71584608-61550400-2b24-11ea-8cde-72a937dd690e.png)
![Screenshot 2019-12-30 at 16 41 56](https://user-images.githubusercontent.com/2330682/71584609-61550400-2b24-11ea-86ad-e1a0ba9b3184.png)
</details>

<details>
  <summary>In compare with current</summary>

![Screenshot 2019-12-30 at 17 04 17](https://user-images.githubusercontent.com/2330682/71585480-c2caa200-2b27-11ea-9056-6bcdde1e9288.png)
![Screenshot 2019-12-30 at 17 04 34](https://user-images.githubusercontent.com/2330682/71585481-c3633880-2b27-11ea-97ad-30b98f801bc9.png)
![Screenshot 2019-12-30 at 17 06 42](https://user-images.githubusercontent.com/2330682/71585482-c3633880-2b27-11ea-9ed8-6d26861db73b.png)
![Screenshot 2019-12-30 at 17 06 51](https://user-images.githubusercontent.com/2330682/71585483-c3633880-2b27-11ea-86f0-164eed06105c.png)
![Screenshot 2019-12-30 at 17 07 01](https://user-images.githubusercontent.com/2330682/71585485-c3633880-2b27-11ea-90f4-4d60acba23c2.png)

</details>

4. Prettier was updated to 1.19.1

umputun commented:
thx for the PR, looking good. One thing I'd suggest to change is a frame color for the active (in-focus) input box. The color you picked (some shade for purple) is not a part of the color scheme used in remark42 anywhere.

<img width="259" alt="fbwwg-201912-30113806-qjfan" src="https://user-images.githubusercontent.com/535880/71593399-7a948b00-2af9-11ea-8ac8-ff965f9cf95d.png">

I'd suggest making it some shade of navy (or whatever the color used in remerak42 called) or, maybe a more visible grey border.

akellbl4 commented:
@umputun :) it is default on-focus browser style. I thought about change it. I will try to pick good color for it.

akellbl4 commented:
@umputun I added styles for focuses.

<details>
  <summary>Before</summary>

![Screenshot 2019-12-30 at 22 32 34](https://user-images.githubusercontent.com/2330682/71597897-03d9ab00-2b56-11ea-8ed0-921276f65e2e.png)
![Screenshot 2019-12-30 at 22 32 53](https://user-images.githubusercontent.com/2330682/71597898-03d9ab00-2b56-11ea-9570-bfb8d09e757a.png)
![Screenshot 2019-12-30 at 22 38 00](https://user-images.githubusercontent.com/2330682/71597899-03d9ab00-2b56-11ea-82a1-0cc3c7c00a43.png)
</details>

<details>
  <summary>After</summary>


![Screenshot 2019-12-30 at 22 31 40](https://user-images.githubusercontent.com/2330682/71597916-0f2cd680-2b56-11ea-85cd-30ddd3a8184e.png)
![Screenshot 2019-12-30 at 22 31 46](https://user-images.githubusercontent.com/2330682/71597918-0f2cd680-2b56-11ea-9736-c759a6def517.png)
![Screenshot 2019-12-30 at 22 31 53](https://user-images.githubusercontent.com/2330682/71597919-0fc56d00-2b56-11ea-94ab-ef3c602b0d6d.png)
</details>

akellbl4 commented:
Email verification view:
<img width="277" alt="Screenshot 2019-12-31 at 02 37 28" src="https://user-images.githubusercontent.com/2330682/71605075-4d86bd80-2b77-11ea-8efe-3f4f902f8df3.png">

umputun commented:
thx a lot. the end result looks good to me.

It will be nice if someone can review the change. pinging @Mavrin @Reeywhaar @Guria - if any of you guys have a moment, pls take a look.

akellbl4 commented:
@umputun yeah, I thought about it and forgot. I will return to this PR after a couple days.

P.S. wish you a Happy New Year ;)

akellbl4 commented on frontend/app/components/auth-panel/__anonymous-login-form/auth-panel__anonymous-login-form.tsx:
```diff
@@ -76,10 +79,9 @@ export class AnonymousLoginForm extends Component<Props, State> {
       <form className={className} onSubmit={this.onSubmit}>
-        <input
-          className="auth-panel-anonymous-login-form__input"
+        <UIInput
```
Yeah, I agree with you. But comment has name `input` and I tried to separate simple UI elements by name.

Mavrin replied:
I think will be better to rename `https://github.com/umputun/remark/blob/master/frontend/app/components/input/input.tsx` to `CommentForm`

akellbl4 replied:
Okay, I agree with you. I will do it in this PR.

akellbl4 commented on frontend/index.ejs:
```diff
@@ -1,118 +1,121 @@
+        site_id: 'remark',
+        host: window.location.origin,
+        url: 'https://remark42.com/demo/',
+        components: ['embed', 'counter'],
```
Oh, sorry, just prettier :)

akellbl4 commented on frontend/app/components/ui-button/ui-button.scss:
```diff
@@ -0,0 +1,21 @@
+
+  &:disabled {
+    opacity: 0.4;
+    cursor: not-allowed;
```
Yep, I took this prop from another button in UI.
I can change it to default

akellbl4 commented on frontend/app/components/dropdown/dropdown.tsx:
```diff
@@ -182,17 +183,16 @@ export default class Dropdown extends Component<Props, State> {
-          className={titleClass}
+          onClick={this.onTitleClick}
+          theme={theme || 'light'}
+          mix={['dropdown__title', titleClass]}
```
@Mavrin it is an example of `Array<string | undefined>` for `Mix` type. Sometimes you need to pass additional class name to an array that can be undefined.

Do you know why BEM method was chosen?

Mavrin replied:
Got it. 
I don't known why was chosen BEM.
I like css modules instead of BEM where it is possible. Or something like https://github.com/lttb/reshadow or https://github.com/4Catalyzer/astroturf.

akellbl4 replied:
@Mavrin agree with you, css-modules are really good for any task. Also, I know another one css-in-js like lib without runtime https://github.com/callstack/linaria :)

akellbl4 commented:
@umputun backend checks failed https://github.com/umputun/remark/pull/510/checks?check_run_id=370725388#step:5:270

@Mavrin I fixed all your comments. Can you check it when you will have time for it? (when lgtm will be fixed)
Please help with testing, because these are really big changes.

umputun commented:
> Please help with testing, because these are really big changes.

I have not found anything this PR breaking, which is good. The only thing I noticed - bold and different font on buttons and not sure if I like it. Not a big expert in UI/UX but to me the old font looked better on buttons and they were very visible enough without bold.

akellbl4 commented:
> The only thing I noticed - bold and different font on buttons and not sure if I like it

All buttons have the same font which was used over the entire interface. The old font which was used in buttons at comment form was different from the interface. In other places I used the same front, I just moved it to button style.

akellbl4 commented on frontend/app/components/comment/comment.tsx:
```diff
@@ -84,12 +85,11 @@ export class Comment extends Component<Props, State> {
       scoreDelta: 0,
       cachedScore: props.data.score,
       initial: true,
+      ...this.updateState(),
```
I accidentally committed a fix for warning about `this.setState` at `constructor`.

akellbl4 commented on frontend/app/components/comment/comment.tsx:
```diff
@@ -102,18 +102,19 @@ export class Comment extends Component<Props, State> {
 
   componentWillReceiveProps(nextProps: Props) {
-    this.updateState(nextProps);
+    this.setState(this.updateState(nextProps));
```
A fix for warning about this.setState at `constructor`.

akellbl4 commented on frontend/app/components/comment/comment.tsx:
```diff
@@ -102,18 +102,19 @@ export class Comment extends Component<Props, State> {
-  updateState = (props: Props) => {
-    this.setState({
+  updateState = (props: Props = this.props) => {
+    const newState: Partial<State> = {
```
A fix for warning about this.setState at `constructor`.

akellbl4 commented on frontend/app/components/comment/comment.tsx:
```diff
@@ -102,18 +102,19 @@ export class Comment extends Component<Props, State> {
       cachedScore: props.data.score,
-    });
+      editDeadline: null,
+    };
```
A fix for warning about this.setState at `constructor`.

akellbl4 commented:
@umputun do you have any notes about this PR? I want to merge it before #496 which I am coding now.

umputun commented:
looks like it has a conflict now. Another thing I'm not sure about - is 101 files changed for this PR is really expected?

akellbl4 commented:
@umputun yep, I checked all files twice.
Lots of files were changed because:

- I added new components `button` and `input` and replaced all different buttons to these components (It changed lots of files)
- I renamed component `input` to `comment-form` which is responsible for comment writing (only this change has 23 files,  you can see it in [this commit](https://github.com/umputun/remark/pull/510/commits/f0ac94c313d40e578319cbd2b2d858ba57e71ed5)).
- I updated `prettier` and rewrote config to JSON ([it updates 9 files](https://github.com/umputun/remark/pull/510/commits/ea31e6cca0d2c1626e7725863da1f892cdbf3ea9) after formatting, [reasons why I moved config to json](https://github.com/umputun/remark/pull/510/commits/2923cf3993b5eeaa3e66dbb01318562977fc6528) in commit comment)

umputun commented:
Visually this PR is fine with me. @Mavrin could you review the final result and give your blessing?

akellbl4 commented:
@Mavrin I need your blessing for this PR. I tried not to break anything :)

akellbl4 commented:
@umputun @Mavrin thanks a lot!

umputun commented:
thank you! The version with nice inputs and buttons were already auto-deployed to the demo site https://remark42.com/demo/
